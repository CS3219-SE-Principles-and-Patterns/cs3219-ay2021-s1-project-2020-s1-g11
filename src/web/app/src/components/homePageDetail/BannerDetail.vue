<template>
  <div>
    <zoom-center-transition :duration="500">      
      <el-row>
        <div v-show="show">
          <h2 class="hidden-md-and-up title-mobile"> 
            ChairVisE
          </h2>
          <el-col :span="6">
            <img alt="Vue logo" src="@/assets/chair2.png"/>
          </el-col>
          <el-col class="hidden-sm-and-down" :span="18">
            <h2> 
              ChairVisE - The Conference Data Visualisation Management System
            </h2>
          </el-col>
        </div>
      </el-row>
    </zoom-center-transition>
  </div>
</template>

<script>
  import {ZoomCenterTransition} from 'vue2-transitions'
  export default {
    name: 'BannerDetail',
    data: () => ({
      show: false,
    }),
    props: {
      msg: String
    },
    mounted() {
      this.loadBanner();
    },
    methods: {
      loadBanner() {
        this.show = true;
      }
    },
    components: {
      ZoomCenterTransition
    }
  }
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
  img {
    width: 250px;
    height: 300px;
    text-align: center;
    display: block;
    margin: 0 auto;
  }
  .el-row {    
    background: #1e9fc4;
    padding: 32px;
  }

  .title-mobile {
    position: absolute;
    margin-left: 180px;
  }
</style>
