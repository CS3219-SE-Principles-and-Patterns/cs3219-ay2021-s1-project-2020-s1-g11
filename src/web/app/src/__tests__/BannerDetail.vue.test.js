import {createLocalVue, shallowMount} from '@vue/test-utils'
import ElementUI from 'element-ui';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import BannerDetail from "./../components/homePageDetail/BannerDetail.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);
localVue.use(ElementUI);

describe("BannerDetail.vue", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        isPageLoading: false,
        userInfo: {
          isApiError: false,
        },
        dbMetaData: {
          entitiesStatus: {
            isLoading: false,
          }
        }
      }
    })
  });

  it('renders correctly', () => {
    const wrapper = shallowMount(BannerDetail, {store, localVue})
    expect(wrapper.element).toMatchSnapshot()
  });
});