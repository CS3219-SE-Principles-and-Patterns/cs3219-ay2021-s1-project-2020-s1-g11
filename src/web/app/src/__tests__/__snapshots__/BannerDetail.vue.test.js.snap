// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerDetail.vue renders correctly 1`] = `
<div>
  <zoom-center-transition-stub
    delay="0"
    duration="500"
    origin=""
    styles="[object Object]"
    tag="span"
  >
    <el-row-stub
      align="top"
      justify="start"
      tag="div"
    >
      <div
        style="display: none;"
      >
        <h2
          class="hidden-md-and-up title-mobile"
        >
           
          ChairVisE
        
        </h2>
         
        <el-col-stub
          span="6"
          tag="div"
        >
          <img
            alt="Vue logo"
            src="@/assets/chair2.png"
          />
        </el-col-stub>
         
        <el-col-stub
          class="hidden-sm-and-down"
          span="18"
          tag="div"
        >
          <h2>
             
            ChairVisE - The Conference Data Visualisation Management System
          
          </h2>
        </el-col-stub>
      </div>
    </el-row-stub>
  </zoom-center-transition-stub>
</div>
`;
